### ISSUE TITLE
pivot_table function implementation

### ISSUE THREAD

**Issue by aregm** [modin-project/modin#1200]

**Comment by dchigarev** [comment]
**Current state:**
All functionality of `pivot_table` were implemented in #1669 (ready for review). Implementation uses `unstack` function, which is not merged in master yet (#1649, `unstack` requires design documentation), so for now it uses `default_to_pandas` version of `unstack`. 

More details about the current state could be found in [this comment](https://github.com/modin-project/modin/pull/1669#issuecomment-662920332).

**Comment by dchigarev** [comment]
**Current state:**
Tested `pivot_table` with draft `unstack` implementation from #1649 here is the results:
```
150000 rows, 256 cols, 256 cols used as values:
pd: 5.6312
md: 4.6227
```
